### PR TITLE
Share button

### LINKED ISSUE #69: Add "share this app" functionality
Should use native share sheets - look into Ionic's [Social Sharing](https://ionicframework.com/docs/native/social-sharing) and [Action Sheet](https://ionicframework.com/docs/native/action-sheet) components. Should be a modular component that can be reused in the app

### PR DESCRIPTION
**Pull request by theswerd**
This is a simple Ionic Share button that shares using @ionic-native/social-sharing.

Resolves #69 

TODO: 

- [ ]  come up with real string to be shared
- [ ] place in proper place within app
- [ ] style with constant color

### CODE REVIEW

File: client/ionic/src/components/ShareButton.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { useState } from "react";
+import { IonContent, IonButton } from "@ionic/react";
+import { SocialSharing } from '@ionic-native/social-sharing';
+
+
+interface ShareConfig {
+    shareSTR: string;
```
Comment (hspinks) [inline client/ionic/src/components/ShareButton.tsx]: super nitpicky code style comment (which brings up the point - we should decide as a group how much we want to standardize and enforce), but I usually prefer camelcase naming (eg `shareStr`). I don't want to impose my own preference yet, but we should decide on standards...

File: client/ionic/src/components/ShareButton.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const ShareButton: React.FC<ShareConfig> = ({ shareSTR }) => {
+    function share(){
+        SocialSharing.share(shareSTR)
+    }
+    return (
+        <IonContent>
+             <IonButton
+              onClick={share}
+              expand="block"
+            >
```
Comment (hspinks) [inline client/ionic/src/components/ShareButton.tsx]: another nitpicky one here, but please just do a quick check of your indentation across the file. The Prettier / Editorconfig changes in progress will help in the future

File: client/ionic/android/app/src/main/res/xml/config.xml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version='1.0' encoding='utf-8'?>
 <widget version="1.0.0" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
-  <!-- TODO: Set access origin: https://github.com/WorldHealthOrganization/app/issues/55 -->
```
[the remark this answers is not in the document]
Comment (theswerd) [reply client/ionic/android/app/src/main/res/xml/config.xml]: To confirm I should leave the todo on line 3?

**Comment by theswerd** [comment]
@hspinks I have no idea why the build is failing now

**Comment by hspinks** [comment]
hmm @theswerd your test output is [here](https://github.com/WorldHealthOrganization/app/pull/73/checks?check_run_id=521153272), but for future reference you can try this locally by running `npm test` from within the `client/ionic` directory to save pushing commits.

the problem is with importing the `@ionic-native/social-sharing` library - when compiling that library, jest finds a JS file using ES6 module imports when it's expecting CommonJS. I think the current Ionic setup is transpiling TS files before running the test suite, but probably not JS files (jest can have some wonkiness when mixing TS and JS). start the debugging process there, see if you can confirm, and if so, see if you can configure jest to transpile JS files too. I can take a deeper look at this tomorrow

File: client/ionic/src/components/ShareButton.tsx (hunk 1 above)
Comment (SamMousa) [reply client/ionic/src/components/ShareButton.tsx]: Wait for #82

**Comment by epicfaace** [comment]
@theswerd to make sure jest transpiles node modules, maybe you can try to create a `jest.config.js` in the same dir as `package.json` with something like this in it:

```json
{
    "transformIgnorePatterns": [
      "node_modules/(?!@ionic-native)"
    ],
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ]
  }
}
```

**Comment by theswerd** [comment]
@hspinks I couldn't get it working

**Comment by theswerd** [comment]
Here is the error:

import { IonicNativePlugin, cordova } from '@ionic-native/core';
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      2 | import { IonContent, IonButton } from "@ionic/react";
      3 | 
    > 4 | import { SocialSharing } from '@ionic-native/social-sharing';
        | ^
      5 | 
      6 | interface ShareConfig {
      7 |   shareStr: string;

      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:537:17)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:579:25)
      at Object.<anonymous> (src/components/ShareButton.tsx:4:1)

**Comment by theswerd** [comment]
@hspinks I am closing as I got a working branch from a fresh start

**Pull request closed without merging**